[inductor] Install guards on both cases of View.handle_negative_index

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103780

This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @ngimel @yf225 @chenyang78

peterbell10 committed:
[inductor] Install guards on both cases of View.handle_negative_index

This branch is not an optimization, it's a correctness issue so there should be
a guards installed for both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Fix small error on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

ezyang reviewed (approved):
A test would be much appreciated

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge

peterbell10 committed:
Update on "[inductor] Install guards on both cases of View.handle_negative_index"


This branch is not an optimization, it's a correctness issue so there should be
a guard installed on both sides of the branch. Otherwise we could have an
expression like `(s0 - s1)` that is initially positive, then becomes negative
with a new set of shapes and now references an invalid index.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge -f "Bazel failure is pre-existing"
